src/bin/ion/commands/beta/inspect.rs

```diff
@@ -344,7 +281,35 @@ impl<'input> IonInspector<'input> {
         // appear each time some number of values is skipped.
         let mut bytes_skipped_this_level = 0;
 
-        while let Some((ion_type, _is_null)) = self.reader.next()? {
+        loop {
+            let (ion_type, _is_null) = match self.reader.next()? {
```

Comment:
Now that we're using a `SystemReader`, we have to distinguish between a few types of data that might appear each time that we call next:
* `None`: No more data at this level of nesting.
* `Some(VersionMarker(_))`: We found an Ion Version Marker.
* `Some(SymbolTableData(_))`: We found an Ion value that was part of a Local Symbol Table (LST) and is not user data.
* `Some(Value(_))`: We found user-level data.

The handling for the last two types is identical; in either case we want to display the bytes that were used to encode that particular value, whether it's user-level or not.